### PR TITLE
Add cell for IdentityManager

### PR DESCRIPTION
**Pull request by christarazi**
- **identitymanager: Add cell for IdentityManager**
- **daemon, identitymanager: Wire up IdentityManager cell to Daemon**

### CODE REVIEW

**Review by dylandreimerink — APPROVED** [review]
LGTM

File: daemon/cmd/identity.go [hunk 1]
```diff
@@ -86,3 +85,13 @@ type CachingIdentityAllocator interface {
...
 	Close()
 }
+
+func (d *Daemon) AddIdentity(id *identity.Identity) {
+	d.idmgr.Add(id)
+}
+func (d *Daemon) RemoveIdentity(id *identity.Identity) {
+	d.idmgr.Remove(id)
+}
+func (d *Daemon) SwapOldNewIdentity(old, new *identity.Identity) {
```
Comment (dlapcevic) [inline daemon/cmd/identity.go]: nit: Any reason to keep the name inconsistent here?
Can it be `RemoveOldAddNewIdentity`?

**Review by dlapcevic — APPROVED** [review]
lgtm

We're planning to have related changes for id allocator very soon, to complete the operator managing cilium identities implementation. We're about to finish all the work in cilium-operator and then follow up with cilium-agent changes.

What do you suggest would be a good way for us to track changes related to our ongoing work?

Identity management is part of sig-policy, but our work was not identified to be entirely relevant to it according to the discussion here: https://github.com/cilium/community/pull/139
Would it have more sense to somehow separately track identity management work or we would want to track and be included in sig-policy?

cc @ovidiutirla

File: daemon/cmd/identity.go (hunk 1 above)
Comment (christarazi) [reply daemon/cmd/identity.go]: Renamed

**Comment by dlapcevic** [comment]
Hi, is this tracked as part of a larger effort for modularization?
Is there any related subsequent work that is planned?

**Comment by christarazi** [comment]
> Hi, is this tracked as part of a larger effort for modularization? Is there any related subsequent work that is planned?

It is part of a large effort for modularization, but when I submitted this PR, I did not have other plans for subsequent work. I just came upon this code and thought I'd improve it while I'm looking at it.

**Comment by dlapcevic** [comment]
> > Hi, is this tracked as part of a larger effort for modularization? Is there any related subsequent work that is planned?
> 
> It is part of a large effort for modularization, but when I submitted this PR, I did not have other plans for subsequent work. I just came upon this code and thought I'd improve it while I'm looking at it.

I see. Thanks

**Review by squeed — APPROVED** [review]
Nice little cleanup, thanks.

**Pull request merged**